R/compat-name-repair.R

```diff
@@ -28,11 +28,12 @@ set_minimal_names <- function(x) {
 unique_names <- function(name, quiet = FALSE, transform = identity) {
   min_name <- minimal_names(name)
   naked_name <- strip_pos(min_name)
-  naked_is_empty <- (naked_name == "")
+  naked_needs_suffix <- (naked_name %in% c("", "..."))
```

Comment:
Can you refresh my memory why `"..."` got added here as a special case that is always considered a duplicate?

Reply:
We need/want to deal with the weird status of `...` which cannot be used as a name even when quoted. This seems to be the only remaining case where the `"unique"` repair gives a non-quotable name. We could describe as:

> The name `...` is special because it is not a valid name even when backticks are used. Just like an empty name it will be "repaired" by appending two dots and a number. The ugliness is intentional.

``` r
tibble::as_tibble(list("..." = 5), .name_repair = "unique")
#> New names:
#> * `...` -> ......1
#> # A tibble: 1 x 1
#>   ......1
#>     <dbl>
#> 1       5
```

<sup>Created on 2019-03-06 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

Reply:
Can you remind me of the example that motivated you? Because the ones I'm exploring here aren't really a slam dunk for why `"..."` has to be repaired for `unique` names. Only `dplyr::select()` seems to balk at it.

``` r
library(tidyverse)

df <- tibble(a = 1)
names(df) <- "..."
df$`...`
#> [1] 1
df[["..."]]
#> [1] 1
select(df, "...")
#> # A tibble: 1 x 1
#>     ...
#>   <dbl>
#> 1     1
select(df, `...`)
#> Error in eval(expr, envir, enclos): '...' used in an incorrect context
```

<sup>Created on 2019-03-06 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

Reply:
``` r
library(tidyverse)

tibble(... = 5) %>%
  mutate_all(list(~ . + 1))
#> Error: '...' used in an incorrect context
```

<sup>Created on 2019-03-06 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

Reply:
Why don't we just treat it exactly like `""` so it becomes `...1` etc?

Reply:
So it it just ... dplyr functions that use tidy eval that barf on `...`? I don't object to repairing them for `unique`, I just want to get the rationale straight.

Reply:
> Why don't we just treat it exactly like `""` so it becomes `...1` etc?

I can't think of a reason NOT to do this. I like it.

Reply:
I think it's the scoped functions that have a problem here. Repairing to `...#` now.